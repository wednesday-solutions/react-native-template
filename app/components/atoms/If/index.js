/**
 *
 * If
 *
 */
// eslint-disable-next-line
import React from 'react'
import Proptypes from 'prop-types';
const If = props => (props.condition ? props.children : props.otherwise);
If.propsTypes = {
  condition: Proptypes.bool,
  otherwise: Proptypes.oneOfType([
    Proptypes.arrayOf(Proptypes.node),
    Proptypes.node
  ]),
  children: Proptypes.oneOfType([
    Proptypes.arrayOf(Proptypes.node),
    Proptypes.node
  ])
};
If.defaultProps = {
  otherwise: null
};
export default If;
