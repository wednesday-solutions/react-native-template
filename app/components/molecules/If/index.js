/**
 *
 * If
 *
 */
// eslint-disable-next-line
import PropTypes from 'prop-types';

const If = props => (props.condition ? props.children : props.otherwise);

If.propTypes = {
  condition: PropTypes.bool,
  otherwise: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]),
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
};

If.defaultProps = {
  otherwise: null
};

export default If;
