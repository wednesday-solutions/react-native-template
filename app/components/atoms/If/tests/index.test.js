/**
 *
 * Tests for If
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { Text } from 'react-native';
import { render } from '@testing-library/react-native';
import If from '../index';

describe('<If />', () => {
  it('Should render and match the snapshot', () => {
    const baseElement = render(<If />);
    expect(baseElement).toMatchSnapshot();
  });

  it('If should render the child component when props.condition = true', () => {
    const conditionTrueText = 'Should render when condition is true';
    const conditionFalseText = 'Should render condition is false';
    const OtherwiseComponent = () => <Text>{conditionFalseText}</Text>;
    const TrueConditionComponent = () => <Text>{conditionTrueText}</Text>;
    const props = {
      otherwise: <OtherwiseComponent />,
      condition: true
    };
    const { queryByText, getByText } = render(
      <If {...props}>
        <TrueConditionComponent />
      </If>
    );
    expect(getByText(conditionTrueText)).toBeTruthy();
    expect(queryByText(conditionFalseText)).not.toBeTruthy();
  });

  it('If should render the component passed on otherwise when props.condition = false', () => {
    const conditionFalseText = 'Should render condition is false';
    const conditionTrueText = 'Should render when condition is true';
    const TrueConditionComponent = () => <Text>{conditionTrueText}</Text>;
    const OtherwiseComponent = () => <Text>{conditionFalseText}</Text>;
    const props = {
      otherwise: <OtherwiseComponent />,
      condition: false
    };

    const { queryByText, getByText } = render(
      <If {...props}>
        <TrueConditionComponent />
      </If>
    );
    expect(getByText(conditionFalseText)).toBeTruthy();
    expect(queryByText(conditionTrueText)).not.toBeTruthy();
  });
});
