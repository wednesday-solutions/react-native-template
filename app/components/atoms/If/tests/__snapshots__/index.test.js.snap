// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<If /> Should render and match the snapshot 1`] = `null`;
